include/deal.II/base/mpi_consensus_algorithms.templates.h

```diff
@@ -557,22 +559,26 @@ namespace Utilities
       Selector<T1, T2>::Selector(Process<T1, T2> &process, const MPI_Comm &comm)
         : Interface<T1, T2>(process, comm)
       {
-        // Depending on the number of processes we switch between
-        // implementations. We reduce the threshold for debug mode to be able to
-        // test also the non-blocking implementation. This feature is tested by:
-        // tests/multigrid/transfer_matrix_free_06.with_mpi=true.with_p4est=true.with_trilinos=true.mpirun=15.output
+        if (Utilities::MPI::job_supports_mpi())
+          {
+            // Depending on the number of processes we switch between
+            // implementations. We reduce the threshold for debug mode to be
+            // able to test also the non-blocking implementation. This feature
+            // is tested by:
+            // tests/multigrid/transfer_matrix_free_06.with_mpi=true.with_p4est=true.with_trilinos=true.mpirun=15.output
 #ifdef DEAL_II_WITH_MPI
 #  if DEAL_II_MPI_VERSION_GTE(3, 0)
 #    ifdef DEBUG
-        if (Utilities::MPI::n_mpi_processes(comm) > 14)
+            if (Utilities::MPI::n_mpi_processes(comm) > 14)
 #    else
-        if (Utilities::MPI::n_mpi_processes(comm) > 99)
+            if (Utilities::MPI::n_mpi_processes(comm) > 99)
 #    endif
-          consensus_algo.reset(new NBX<T1, T2>(process, comm));
-        else
+              consensus_algo.reset(new NBX<T1, T2>(process, comm));
+            else
```

Comment:
and then use here
```suggestion
            else if (n_procs > 1)
```

Reply:
Which would then allow you to limit the `job_supports_mpi` to a single place, the constructor of `Interface`. It would have the side effect that we do not try to set up anything for the single-rank case even if we are inside MPI.

Reply:
Done!